Add rename ingest processor (#6951)
opensearch-project/documentation-website: _ingest-pipelines/processors/rename.md

---
layout: default
title: Rename
parent: Ingest processors
nav_order: 230
redirect_from:
   - /api-reference/ingest-apis/processors/rename/
---

# Rename processor

> The `rename` processor is used to rename an existing field, which can also be used to move a field from one object to another object or the root level.

Reviewer: ```suggestion
The `rename` processor is used to rename an existing field, which can also be used to move a field from one object to another object or to the root level.
```

## Syntax

The following is the syntax for the `rename` processor: 

```json
{
    "rename": {
        "field": "field_name",
        "target_field" : "target_field_name"
    }
}
```
{% include copy-curl.html %}

## Configuration parameters

The following table lists the required and optional parameters for the `rename` processor.

| Parameter  | Required/Optional  | Description  |
|---|---|---|
`field`  | Required  | The field name containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
`target_field`  | Required  | The new name of the field. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
`ignore_missing`  | Optional  | Specifies whether the processor should ignore documents that do not contain the specified `field`. If set to `true`, the processor does not modify the document if the `field` does not exist. Default is `false`. |
`description`  | Optional  | A brief description of the processor.  |
`if` | Optional | A condition for running the processor. |

> `ignore_failure` | Optional | Specifies whether the processor continues execution even if it encounters errors. If set to `true`, failures are ignored. Default is `false`. |

Reviewer: ```suggestion
`ignore_failure` | Optional | Specifies whether the processor continues execution even if it encounters an error. If set to `true`, failures are ignored. Default is `false`. |
```

`on_failure` | Optional | A list of processors to run if the processor fails. |
`tag` | Optional | An identifier tag for the processor. Useful for debugging in order to distinguish between processors of the same type. |

## Using the processor

Follow these steps to use the processor in a pipeline.

**Step 1: Create a pipeline** 

> The following query creates a pipeline named `rename_field` that moves a field in a object to the root level: 

Reviewer: ```suggestion
The following query creates a pipeline named `rename_field` that moves a field in an object to the root level: 
```
